emperor/util.py

```diff
@@ -652,3 +654,46 @@ def guess_coordinates_files(dir_path):
         coord_fps.append(filepath)
 
     return coord_fps
+
+
+def validate_and_process_custom_axes(mf, custom_axes):
+    """Validate and process mapping file for custom axes
+
+    Parameters
+    ----------
+    mf : pd.DataFrame
+        The sample metadata.
+    custom_axes : list of str
+        The custom axes to extract from the metadata.
+
+    Returns
+    -------
+    pd.DataFrame
+        A resulting DataFrame with the custom axes as numeric types.
+
+    Raises
+    ------
+    KeyError
+        When the category names in `custom_axes` are not present in `mf`.
+    ValueError
+        When there are non-numeric values in the categories selected by
+        `custom_axes`.
+    """
+    # avoid side-effects
+    mf = mf.copy()
+
+    headers = [mf.index.name] + mf.columns.tolist()
+    missing_headers = set(custom_axes).difference(set(headers))
+
+    if missing_headers:
+        raise KeyError("One or more headers are not present in the "
+                       "sample information: %s" % ', '.join(missing_headers))
+
+    for axis in custom_axes:
+        mf[axis] = pd.to_numeric(mf[axis], errors='coerce')
```

Comment:
This might be a pain to support and may be considered out of scope, but it might be cool to allow lexicographic sort. It would also allow for the case where the sample values are like `{"01.daily", "02.weekly", "03.monthly", "04.biannually", "05.annually"}` where the advantage is that the values associated with the metadata have meaning while using `{1..5}` in this context would potentially be confusing